tests/components/unifi/test_update.py

```diff
@@ -122,46 +136,79 @@ async def test_device_updates(
     assert device_1_state.attributes[ATTR_IN_PROGRESS] is False
 
 
+async def test_not_admin(hass, aioclient_mock):
+    """Test that the INSTALL feature is not available on a non-admin account."""
+    description = deepcopy(DESCRIPTION)
+    description[0]["site_role"] = "not admin"
+
+    await setup_unifi_integration(
+        hass,
+        aioclient_mock,
+        site_description=description,
+        devices_response=[DEVICE_1],
+    )
+
+    assert len(hass.states.async_entity_ids(UPDATE_DOMAIN)) == 1
+    device_state = hass.states.get("update.device_1")
+    assert device_state.state == STATE_ON
+    assert (
+        device_state.attributes[ATTR_SUPPORTED_FEATURES] == UpdateEntityFeature.PROGRESS
+    )
+
+
+async def test_install(hass, aioclient_mock):
+    """Test the device update install call."""
+    config_entry = await setup_unifi_integration(
+        hass, aioclient_mock, devices_response=[DEVICE_1]
+    )
+
+    assert len(hass.states.async_entity_ids(UPDATE_DOMAIN)) == 1
+    device_state = hass.states.get("update.device_1")
+    assert device_state.state == STATE_ON
+
+    controller = hass.data[UNIFI_DOMAIN][config_entry.entry_id]
```

Comment:
Please don't access integration details like hass.data in the tests. Don't we know the controller host and site already since we set up the integration?

Reply:
I've changed this in #75348.